core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -6418,6 +6430,17 @@ public SqlNode expandGroupByOrHavingOrQualifyExpr(SqlNode expr,
     return newExpr;
   }
 
+  public SqlNode expandWithAlias(SqlNode expr,
```

Comment:
Can you a docstring here since this seems to be the real implementation that's used?